irepela/homework_11/guess_word.py

```diff
@@ -0,0 +1,122 @@
+import argparse
+import random
+import csv
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('-p', '--path',
+                        action='store',
+                        dest='path',
+                        help='path to a file with tasks')
+
+    parser.add_argument('-s', '--sep',
+                        action='store',
+                        default="|",
+                        dest='sep',
+                        help='separator for file with tasks')
+
+    return parser.parse_args()
+
+
+class Message:
+    INVALID = "Wrong value '{guess}', try again"
+    WRONG = "Wrong, try again"
+    WIN = "You won"
+    CORRECT = "Correct"
+    INPUT_REQUEST = "Make a guess (one letter): "
+
+
+def get_task(args) -> tuple:
+    """
+    Function which reads tasks from a file and gets random task
+    """
+    path = args.path
+    sep = args.sep
+
+    with open(path) as csvfile:
+        file_tasks = list(csv.reader(csvfile, delimiter=sep))
+    file_task, file_desc = random.choice(file_tasks)
+
+    return file_task, file_desc
+
+
+class ScreenView:
+
+    def __init__(self, session: "GameSession"):
+        self.session = session
+
+    def show_start_screen(self):
+        print(f"Word description: {self.session.desc}")
+        mask = ["*" for _ in self.session.task]
+        print("   ".join(mask))
+
+    @staticmethod
+    def fire_alert(message: str):
+        print(message)
+
+    def show_current_state(self):
+        mask = list("*" * (len(self.session.task) - len(self.session.tries)))
+        print("   ".join(self.session.tries + mask))
+
+
+class GameSession:
+
+    ALLOWED_CHARS = "abcdefghijklmnopqrstuvwxyz"
+
+    def __init__(self, task: str, desc: str):
+        self.task = task
+        self.desc = desc
+        self.tries = []
+        self.screen = ScreenView(self)
+
+    @staticmethod
+    def user_make_guess() -> str:
+        """
+        Receive input from user, normalize
+        """
+        return input(Message.INPUT_REQUEST).strip().lower()
+
+    def win(self) -> bool:
+        """
+        Check if all letters are guessed
+        """
+        return self.task == "".join(self.tries)
```

Comment:
you wil never win since you need to gues in same order as letters in answer, and make no mistakes